python_backend/database/storage.py

```
@@ -209,4 +210,172 @@ def save_negative_preferences(self, user_id: str, preferences: Dict[str, Any]) -
+                        embedding_model=metadata['embedding_model'],
+                        chunk_count=metadata['chunk_count'],
+                        chunk_size=metadata['chunk_size'],
+                        chunk_overlap=metadata['chunk_overlap'],
+                        faiss_index_path=metadata['faiss_index_path'],
+                        vector_dimension=metadata['vector_dimension'],
+                        status=metadata.get('status', 'active')
+                    )
+                    db.add(doc_metadata)
+
+                db.commit()
+                return True
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)